### ISSUE TITLE
Turn off tqdm from application code

### ISSUE THREAD

**Issue by stellarhoof** [tqdm/tqdm#619]
- [X] I have visited the [source website], and in particular
  read the [known issues]
- [X] I have searched through the [issue tracker] for duplicates
- [X] I have mentioned version numbers, operating system and
  environment, where applicable:

```
4.26.0 3.6.6 (default, Aug 21 2018, 23:42:39) 
[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)] darwin
```

I'm using a package that uses `tqdm` internally and I'd like to disable tqdm *from within application code*.

https://github.com/tqdm/tqdm/issues/612 would've been ideal. It was closed because the solution solved the problem for the author of the issue but not for us since we cannot modify library code. Thanks.

**Comment by casperdcl** [comment]
before importing the offending package you could do something like:

```python
import tqdm

def nop(it, *a, **k):
    return it

tqdm.tqdm = nop
```

**Comment by stellarhoof** [comment]
That would work but it's not very robust since it depends on the order of imports. It'd be fine if I was doing a one-off script for example where I have absolute control of how I import things but if I monkey patch `tqdm` like you suggested somewhere else to be reused it's going to cause very confusing behavior.

**Comment by thakkarparth007** [comment]
Any updates on this? Seems very useful when I use this in Jupyter notebooks and want to sometimes disable progress bars globally.

**Comment by casperdcl** [comment]
I think the main discussion is #614

**Comment by casperdcl** [comment]
Closing this in favour of #614

**Comment by kaivalyar** [comment]
For those still stumbling upon this problem, perhaps this might be useful, as an improvement of @casperdcl's solutions:

```
import tqdm
class _TQDM(tqdm.tqdm):
    def __init__(self, *argv, **kwargs):
        kwargs['disable'] = True
        if kwargs.get('disable_override') is not None:
            kwargs['disable'] = kwargs['disable_override']
        super().__init__(*argv, **kwargs)
tqdm.tqdm = _TQDM
```

This has almost the identical effect as the `nop` in @casperdcl's solution, and can be used in exactly the same manner. However, this has the added benefit of allowing the use of tqdm in the application code itself. So effectively the behaviour of tqdm when used by library code is to always set `disable` to `True`, but the application code can use tqdm using the new parameter `disable_override`.

This can also be further customised to not override the library parameter settings, but only change tqdm's default behaviour, by replacing the line `kwargs['disable'] = True` with `kwargs['disable'] = kwargs.get('disable', True)`, if desired.

**Comment by AdityaVenkatraman** [comment]
Thank you very much! This is exactly what I needed.

**Comment by ndvbd** [comment]
@kaivalyar  I added these lines to my code - nothing happened - tqdm still shows progress bars.

**Comment by casperdcl** [comment]
Thanks everyone for your patience! Just released [`tqdm==4.66.0`](https://github.com/tqdm/tqdm/releases/tag/v4.66.0) which includes support for env var overrides. I'm using `TQDM_MININTERVAL=5` in most of my CI jobs now :)
